File: backend/models/WeddingDetails.ts
import mongoose from "mongoose";

const Schema = mongoose.Schema;

const WeddingDressSchema = new Schema({
    weddingDressUUID: {
        type: Schema.Types.String,
        required: true
    },
    link: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    },
    photo: {
        type: Schema.Types.String,
        default: ""
    },
    description: {
        type: Schema.Types.String,
        default: ""
    }
});

export const WeddingDress = mongoose.model("weddingDress", WeddingDressSchema)

const BouquetSchema = new Schema({
    bouquetUUID: {
        type: Schema.Types.String,
        required: true
    },
    link: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    },
    photo: {
        type: Schema.Types.String,
        default: ""
    },
    description: {
        type: Schema.Types.String,
        default: ""
    }
});

export const Bouquet = mongoose.model("bouquet", BouquetSchema)

const GroomSuitSchema = new Schema({
    groomSuitUUID: {
        type: Schema.Types.String,
        required: true
    },
    link: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    },
    photo: {
        type: Schema.Types.String,
        default: ""
    },
    description: {
        type: Schema.Types.String,
        default: ""
    }
});

export const GroomSuit = mongoose.model("groomSuit", GroomSuitSchema)

const ChurchCeremonySchema = new Schema({
    churchCeremonyUUID: {
        type: Schema.Types.String,
        required: true
    },
    churchAddress: {
        type: Schema.Types.String,
        default: ""
    },
    date: {
        type: Schema.Types.String,
        default: ""
    },
    hour: {
        type: Schema.Types.String,
        default: ""
    },
    preotName: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    }
});

export const ChurchCeremony = mongoose.model("churchCeremony", ChurchCeremonySchema)

const PartyLocationSchema = new Schema({
    partyLocationUUID: {
        type: Schema.Types.String,
        required: true
    },
    partyAddress: {
        type: Schema.Types.String,
        default: ""
    },
    date: {
        type: Schema.Types.String,
        default: ""
    },
    hour: {
        type: Schema.Types.String,
        default: ""
    },
    decorationsOrganizerDetails: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    }
});

export const PartyLocation = mongoose.model("partyLocation", PartyLocationSchema)

const CivilMarriageSchema = new Schema({
    civilMarriageUUID: {
        type: Schema.Types.String,
        required: true
    },
    address: {
        type: Schema.Types.String,
        default: ""
    },
    date: {
        type: Schema.Types.String,
        default: ""
    },
    hour: {
        type: Schema.Types.String,
        default: ""
    }
});

export const CivilMarriage = mongoose.model("civilMarriage", CivilMarriageSchema)

const FoodMenuSchema = new Schema({
    foodMenuUUID: {
        type: Schema.Types.String,
        required: true
    },
    antreu: [{
        type: Schema.Types.String,
        default: ""
    }],
    firstCourse: [{
        type: Schema.Types.String,
        default: ""
    }],
    mainCourse: [{
        type: Schema.Types.String,
        default: ""
    }],
    secondMainCourse: [{
        type: Schema.Types.String,
        default: ""
    }],
    price: {
        type: Schema.Types.Int32,
        default: 0
    }
});

export const FoodMenu = mongoose.model("foodMenu", FoodMenuSchema)

const BarMenuSchema = new Schema({
    barMenuUUID: {
        type: Schema.Types.String,
        required: true
    },
    alcoholic: [{
        type: Schema.Types.String,
        default: ""
    }],
    nonalcoholic: [{
        type: Schema.Types.String,
        default: ""
    }],
    coffee: [{
        type: Schema.Types.String,
        default: ""
    }],
    juice: [{
        type: Schema.Types.String,
        default: ""
    }],
    price: {
        type: Schema.Types.Int32,
        default: 0
    }
});

export const BarMenu = mongoose.model("barMenu", BarMenuSchema)

const WeddingCakeSchema = new Schema({
    weddingCakeUUID: {
        type: Schema.Types.String,
        required: true
    },
    name: {
        type: Schema.Types.String,
        default: ""
    },
    photo: {
        type: Schema.Types.String,
        default: ""
    },
    description: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    }
});

export const WeddingCake = mongoose.model("weddingCake", WeddingCakeSchema)

const LiveBandSchema = new Schema({
    liveBandUUID: {
        type: Schema.Types.String,
        required: true
    },
    name: {
        type: Schema.Types.String,
        default: ""
    },
    price: {
        type: Schema.Types.Int32,
        default: 0
    },
    hour: {
        type: Schema.Types.String,
        default: ""
    },
    details: {
        type: Schema.Types.String,
        default: ""
    },
    
});

export const LiveBand = mongoose.model("liveBand", LiveBandSchema)

const WeddingDetailsSchema = new Schema({
    weddingDetailsUUID: {
        type: Schema.Types.String,
        required: true
    },
    date: {
        type: Schema.Types.String,
        default: ""
    },
    weddingDress: {
        type: Schema.Types.ObjectId,
        ref: 'weddingDress'
    },
    bouquet: {
        type: Schema.Types.ObjectId,
        ref: 'bouquet'
    },
    groomSuit: {
        type: Schema.Types.ObjectId,
        ref: 'groomSuit'
    },
    churchCeremony: {
        type: Schema.Types.ObjectId,
        ref: 'churchCeremony'
    },
    partyLocation: {
        type: Schema.Types.ObjectId,
        ref: 'partyLocation'
    },
    civilMarriage: {
        type: Schema.Types.ObjectId,
        ref: 'civilMarriage'
    },
    foodMenu: {
        type: Schema.Types.ObjectId,
        ref: 'foodMenu'
    },
    barMenu: {
        type: Schema.Types.ObjectId,
        ref: 'barMenu'
    },
    weddingCake: {
        type: Schema.Types.ObjectId,
        ref: 'weddingCake'
    },
    liveBand: {
        type: Schema.Types.ObjectId,
        ref: 'liveBand'
    }
})

export const WeddingDetails = mongoose.model("weddingDetails", WeddingDetailsSchema)

File: WeddingPlannerBuddy/Model/WeddingDetails.swift
//
//  WeddingDetails.swift
//  WeddingPlannerBuddy
//
//  Created by Alexia Aldea on 30.01.2025.
//

import Foundation
import SwiftUICore

struct WeddingDetails {
    let id: String
    let date: String
    let price: Int
    let weddingDress: WeddingDress
    let bouquet: Bouquet
    let groomSuit: GroomSuit
    let churchCeremony: ChurchCeremony
    let partyLocation: PartyLocation
    let civilMarriage: CivilMarriage
    let foodMenu: FoodMenu
    let barMenu: BarMenu
    let weddingCake: WeddingCake
    let liveBand: LiveBand
}

struct WeddingDress {
    let id: String
    let link: String
    let price: Int
    let photo: String
    let description: String
}

struct Bouquet {
    let id: String
    let link: String
    let price: Int
    let photo: String
    let description: String
}

struct GroomSuit {
    let id: String
    let link: String
    let price: Int
    let photo: String
    let description: String
}

struct ChurchCeremony {
    let id: String
    let churchAddress: String
    let date: String
    let hour: String
    let preotName: String
    let price: Int
}

struct PartyLocation {
    let id: String
    let partyAddress: String
    let date: String
    let hour: String
    let decorationsOrganizerDetails: String
    let price: Int
}

struct CivilMarriage {
    let id: String
    let address: String
    let date: String
    let hour: String
}

struct FoodMenu {
    let id: String
    let antreu: [String]
    let firstCourse: [String]
    let mainCourse: [String]
    let secondMainCourse: [String]
    let price: Int
}

struct BarMenu {
    let id: String
    let alcoholic: [String]
    let nonalcoholic: [String]
    let coffee: [String]
    let juice: [String]
    let price: Int
}

struct WeddingCake {
    let id: String
    let name: String
    let photo: String
    let description: String
    let price: Int
}

struct LiveBand {
    let id: String
    let name: String
    let price: Int
    let hour: String
    let details: String
}

struct PriceItem {
    let name: String
    let price: Int
    let color: Color
}

struct ScheduleItem {
    let name: String
    let date: String
    let hour: String
}

extension WeddingDetails {
    
    var priceItems: [PriceItem] {
        var items: [PriceItem] = []

        items.append(PriceItem(name: "Wedding dress", price: weddingDress.price, color: Color(hex: "#FF7F3A")))
        items.append(PriceItem(name: "Bouquet", price: bouquet.price, color: Color(hex: "#E85678")))
        items.append(PriceItem(name: "Groom suit", price: groomSuit.price, color: Color(hex: "#37B5EF")))
        items.append(PriceItem(name: "Church ceremony", price: churchCeremony.price, color: Color(hex: "#3DC269")))
        items.append(PriceItem(name: "Party location", price: partyLocation.price, color: Color(hex: "#EB9D34")))
        items.append(PriceItem(name: "Food menu", price: foodMenu.price, color: Color(hex: "#2B2A2C")))
        items.append(PriceItem(name: "Bar menu", price: barMenu.price, color: Color(hex: "#F91E4A")))
        items.append(PriceItem(name: "Wedding cake", price: weddingCake.price, color: Color(hex: "#9400BD")))
        items.append(PriceItem(name: "Live band", price: liveBand.price, color: Color(hex: "#00C2A8")))

        return items
    }

    var scheduleItems: [ScheduleItem] {
        var items: [ScheduleItem] = []

        items.append(
            ScheduleItem(
                name: "Church ceremony",
                date: churchCeremony.date,
                hour: churchCeremony.hour
            )
        )

        items.append(
            ScheduleItem(
                name: "Party location",
                date: partyLocation.date,
                hour: partyLocation.hour
            )
        )

        items.append(
            ScheduleItem(
                name: "Civil marriage",
                date: civilMarriage.date,
                hour: civilMarriage.hour
            )
        )
        
        items.append(
            ScheduleItem(
                name: "Live band",
                date: "",
                hour: liveBand.hour
            )
        )

        return items
    }
}
